### DISCUSSION TITLE
Dextran and Cellulose parameters

### DISCUSSION THREAD

**Discussion by junzhoux** [marrink-lab/polyply_1.0#254] [Q&A] [answered]
I read a new preprint paper that extend the Martini 3 to carbohydrates and it says that the dextran and cellulose parameters have been included into the polyply. When I check the parameters, it seems that the bead types of dextran are different from the descriptions in the paper, and the parameters for cellulose are missing. Sorry to bother you here, I just want to simulate a cellulose fibres system and want to try these new parameters.

**Comment by fgrunewald** [comment, marked as answer]
Hi @junzhoux, indeed you're right that the preprint mentions the parameters are part of polyply and they sort of are. However, they are still part of a [draft pull-request](https://github.com/marrink-lab/polyply_1.0/pull/240). The reasoning is that we don't want to add more preliminary parameters into an official release, as they might undergo changes in the reviewing process of the paper. However, we didn't anticipate how long it would take to receive those reviews. Long story short my apologies for things moving slower than they probably should.

In order to use the latest parameters, you can either checkout the [draft pull-request](https://github.com/marrink-lab/polyply_1.0/pull/240) or get the input files from [here for Cellulose](https://github.com/marrink-lab/polyply_1.0/blob/carbo_perprint/polyply/data/martini3/cellulose.martini3.ff) or [here for Dextran](https://github.com/marrink-lab/polyply_1.0/blob/carbo_perprint/polyply/data/martini3/dextran.martini3.ff). Copy the content or download the file. Then you can generate the itp-files as follows for Cellulose or Dextran respectively. 
```
polyply gen_params -f cellulose.martini3.ff -seq CEL:20 -o cellulose.itp -name CEL
polyply gen_params -f dextran.martini3.ff -seq DEX:20 -o cellulose.itp -name DEX
```
Coordinates for dispersed cellulose chains are generated as for any other polymer, but you might want to tune the persistence length of the random-walk to produce more straight chains. You can find more information on how to tune the persistence length in [this tutorial](https://github.com/marrink-lab/polyply_1.0/wiki/Tutorial:-Single-stranded-circular-DNA). However, in order to build a fiber you probably will have to generate the coordinates with an atomistic builder and then map them as was done in the paper. In case your fiber follows Charmm36m naming convention I can make the Glucose mapping files available to you as well. Let me know.

Finally, any feedback about the parameters in case something goes wrong or you observe unexpected behavior with the model is appreciated. Please also give me a quick heads up if you managed to follow the instructions outlined. 

Cheers,
Fabian

**Reply by junzhoux** [reply]
Hi @fgrunewald, thanks for your informative response, I will try to follow your instructions. In my case, I also need to modify the fibre surface by replacing C6-OH with a sulfate group or a carboxylic group. The atomistic model follows CHARMM36m naming convention, if you can give me the Glucose mapping files, that'll help me a lot. Many thanks.

Best wishes,
Jun

**Comment by fgrunewald** [comment]
Hi @junzhoux, 

Below the mapping file for Charmm36m beta Glucose. 

```
; mapping for beta Galactose
[ molecule ]
BGAL GAL
[ martini ]
A B C V
[ atoms ]
     1     C1      A   V
     2     H1      A   V
     3     O1      A   V
     4    HO1      A   V
     5     C5      C   V
     6     H5      C   V
     7     O5      C   V
     8     C2      A   V
     9     H2      A   V
    10     O2      A   V
    11    HO2      A   V
    12     C3      B   V
    13     H3      B   V
    14     O3      B   V
    15    HO3      B   V
    16     C4      B   V
    17     H4      B   V
    18     O4      B   V
    19    HO4      B   V
    20     C6      C   V
    21    H61      C   V
    22    H62      C   V
    23     O6      C   V
    24    HO6      C   V
```

Carboxylic acid groups attached to carbohydrates, we have parametrised in the paper as well. You should be able to figure out how the bead types change from Figure 1 & 4 of the preprint. In principle that should also work for sulfate groups given you determine the bead-type for those from the standard rules. If you're in doubt feel free to ask here as well.

**Reply by junzhoux** [reply]
Hi @fgrunewald,  I used to deal with the small surfactant molecule, and have not mapped such big molecules before. I also build the initial structure of cellulose fibre using cellulose-builder, can you please show me how to use that mapping file, do I need to use martinize?  

Regards,

**Reply by fgrunewald** [reply]
Martinez will not be able to map the structure because of missing input files. You can either use gmx trjconv and convert the mapping I sent to an index file. Alternatively, we have this program https://github.com/fgrunewald/fast_forward, which does it from scratch if you can provide an all-atom tar file. However, fast_forward is still under development so no guarantees. For the paper we converted the mapping file to an index file and used gmx trjconv I believe.

**Reply by junzhoux** [reply]
Thanks for your suggestions, I have successfully made a coarse-grained model. Regarding the parameterization of sulfated glucose, I am just wondering if it is necessary to assign a dihedral constraint to the sulfated group bead and the ring. I have tried several times, the dihedral angle distribution between AA results and CG results is very different.

**Reply by fgrunewald** [reply]
Great! Good to hear that it worked out so far. If the conformations are very different you can certainly have a dihedral constraint or improper dihedral angle. If the orientation is important to your work you can also try constructing it as a virutal site from the ring beads. The solution depends on what you need from your model. But be warned including dihedral angles is a difficult task and you don't always find a perfect solution.

**Reply by junzhoux** [reply]
Hi @fgrunewald, I have another question. The cellulose contains 36 chains, but polyply can generate the topology of a single chain, how do you extend it to cover the 36 chains?

**Reply by fgrunewald** [reply]
Are those chains equivalent? If yes you would just use the same itp file for all chains. If not then you need to run polyply on each distinct chain.

**Reply by junzhoux** [reply]
Hi @fgrunewald, I have another question about the mapping of cellulose. For the CEL.itp file (polyply/tests/test_data/library_tests/martini3/CEL/polyply/), Why there is a bond between 2 (B/SP4r) and 5(A/SN6) beads? I think there should be a bond between type SP1r and SN6?? I am not sure if I made a mistake, please have a look.

**Reply by fgrunewald** [reply]
@junzhoux Good catch! There was a small mistake in the input files for Cellulose. They only affect the very first and last residue. In fact the bond was correct but the bead assignments were incorrect. The A bead in the first residue should be SP4r and the B bead SP1r. Bonds for cellulose run from B->A by the convention used in our paper. Then of course the middle is not affected because the substitution repeats, but the last residue is also incorrect. I don't think it affects any physical properties of Cellulose but of course I fixed it. 

Can you try to generate the strand again and see if it is fixed for you? Note you need to update the branch using `git pull`

**Reply by junzhoux** [reply]
Hi @fgrunewald, do you mean I need to modified the first and last residue manually?

**Reply by fgrunewald** [reply]
Hi @junzhoux: Nope, I mean that the code is now fixed and will produce the correct itp file after you updated your local version. Simply copy the input file [from here again](https://github.com/marrink-lab/polyply_1.0/blob/7782f36a492ccd50b9a9424ad8f6b597ae4edcec/polyply/data/martini3/cellulose.martini3.ff) it should be working is what I mean.
